include/geom/bounding_box.h

```diff
@@ -147,6 +147,60 @@ class BoundingBox : public std::pair<Point, Point>
 // ------------------------------------------------------------
 // BoundingBox class member functions
 
+inline
+bool
+BoundingBox::intersects (const BoundingBox & other_box) const
+{
+  const libMesh::Point & my_lower = this->first;
+  const libMesh::Point & my_upper = this->second;
+
+  const libMesh::Point & other_lower = other_box.first;
+  const libMesh::Point & other_upper = other_box.second;
+
+  // Since boxes are tensor products of line intervals it suffices to check
+  // that the line segments for each coordinate axis overlap.
+  for (unsigned int dir=0; dir<LIBMESH_DIM; ++dir)
+    {
+      // Line segments can intersect in two ways:
+      // 1. They can overlap.
+      // 2. One can be inside the other.
+      //
+      // In the first case we want to see if either end point of the second
+      // line segment lies within the first. In the second case we can simply
+      // check that one end point of the first line segment lies in the second
+      // line segment. Note that we don't need, in the second case, to do two
+      // checks since that case is already covered by the first.
+      if (!((my_lower(dir) <= other_lower(dir) &&
+             other_lower(dir) <= my_upper(dir)) ||
+            (my_lower(dir) <= other_upper(dir) &&
+             other_upper(dir) <= my_upper(dir))) &&
+          !((other_lower(dir) <= my_lower(dir) &&
+             my_lower(dir) <= other_upper(dir))))
+        {
+          return false;
+        }
+    }
+
+  return true;
+}
+
+inline
+bool
+BoundingBox::intersects (const BoundingBox & other_box,
+                         Real abstol) const
+{
+  BoundingBox expanded_my_box = *this;
+  BoundingBox expanded_other_box = other_box;
+  for (unsigned int dir=0; dir<LIBMESH_DIM; ++dir)
+    {
+      expanded_my_box.first(dir) -= abstol;
+      expanded_my_box.second(dir) += abstol;
+      expanded_other_box.first(dir) -= abstol;
+      expanded_other_box.second(dir) += abstol;
+    }
```

Comment:
Should we only be expanding one box or the other?  I think that'll match the previous abstol behavior better, and it'll save a few flops.

Reply:
You're absolutely right: I'll fix this.

Reply:
Weird; I'd have expected Github to have marked this conversation as outdated after the force push.